jetty-quickstart/src/main/java/org/eclipse/jetty/quickstart/QuickStartConfiguration.java

```diff
@@ -72,57 +73,70 @@ public static void configureMode(Server server, String mode)
     public enum Mode
     {
         GENERATE,  // Generate quickstart-web.xml and then stop
-        AUTO,      // use or generate depending on the existance of quickstart-web.xml
+        AUTO,      // use or generate depending on the existence of quickstart-web.xml
         QUICKSTART // Use quickstart-web.xml
     }
 
-    private Mode _mode = Mode.AUTO;
-    private boolean _quickStart;
-
     public QuickStartConfiguration()
     {
         super(true);
         addDependencies(WebInfConfiguration.class);
         addDependents(WebXmlConfiguration.class);
     }
 
+    private Mode getModeForContext(WebAppContext context)
+    {
+        Mode mode = (Mode)context.getAttribute(MODE);
+        if (mode == null)
+            return DEFAULT_MODE;
+        return mode;
+    }
+
     @Override
     public void preConfigure(WebAppContext context) throws Exception
     {
-        //check that webapp is suitable for quick start - it is not a packed war
+        // check that webapp is suitable for quick start - it is not a packed war
         String war = context.getWar();
-        if (war == null || war.length() <= 0 || !context.getBaseResource().isDirectory())
-            throw new IllegalStateException("Bad Quickstart location");
+        if (StringUtil.isBlank(war) || !context.getBaseResource().isDirectory())
+            throw new IllegalStateException("Invalid Quickstart location");
 
-        //look for quickstart-web.xml in WEB-INF of webapp
+        // look for quickstart-web.xml in WEB-INF of webapp
         Resource quickStartWebXml = getQuickStartWebXml(context);
-        LOG.debug("quickStartWebXml={} exists={}", quickStartWebXml, quickStartWebXml.exists());
+        if (quickStartWebXml == null)
+        {
+            if (LOG.isDebugEnabled())
+                LOG.debug("No quickStartWebXml present on {}", context);
+            return;
+        }
 
-        //Get the mode
-        Mode mode = (Mode)context.getAttribute(MODE);
-        if (mode != null)
-            _mode = mode;
-        
-        _quickStart = false;
```

Comment:
Let's just add a boolean context attribute that captures if  the context will be quickstarted to replace the `_quickstart` field.

Reply:
We need a new Mode.

Something like `DISABLED` so that someone that has a webapp, that has a `WEB-INF/quickstart-web.xml`, but doesn't want to use QuickStart for that specific webapp, can set the context attribute to this `DISABLED` state.

Reply:
@joakime a new `DISABLED` mode could be added for jetty-12 in another PR, but not in this one for jetty-10, which is not having new features added to it.